lxd/events/connections.go

```diff
@@ -128,3 +132,103 @@ func (e *websockListenerConnection) WriteJSON(event any) error {
 
 	return e.Conn.WriteJSON(event)
 }
+
+// NewStreamListenerConnection returns a new http stream listener connection.
+func NewStreamListenerConnection(connection net.Conn) EventListenerConnection {
```

Comment:
This function needs to return an `error` so it can return an error from the `io.WriteString` which is currently incorrectly ignored.